ir/expression.def

```diff
@@ -267,6 +274,9 @@ class StringLiteral : Literal {
 #emit
     operator IR::ID() const { return IR::ID(srcInfo, value); }
 #end
+
+    /// @returns a string literal. The value is cached.
+    static const StringLiteral *get(cstring value, const Type *t = Type_String::get(), const Util::SourceInfo &si = {});
```

Comment:
The literals with non-string type seem quite niche (or even hacky). I wonder if these should be created by a different overload (or just use the constructor for them). I am not too happy with the `get` functions having different order of arguments then the constructors, that could be quite confusing :-(.

Reply:
There is `Type_MatchKind` which could be use here. I did not want to restrict the type choice too much.